packages/docusaurus/bin/docusaurus.js

```diff
@@ -312,8 +312,19 @@ function isInternalCommand(command) {
 }
 
 async function run() {
-  if (!isInternalCommand(process.argv.slice(2)[0])) {
-    await externalCommand(cli, path.resolve('.'));
+  const commandName = process.argv.slice(2)[0];
+
+  if (!isInternalCommand(commandName)) {
+    try {
+      await externalCommand(cli, path.resolve('.'));
```

Comment:
Not 100% sure but I think this does not really execute the command, it just registers it by loading plugin commands (otherwise we skip this process as an optimisation)

I think `cli.parse(process.argv);` is what executes the actual external command, so it should rather be added to the try/catch

Reply:
btw we should probably use `cli.parseAsync` instead of `cli.parse`

> Use parseAsync instead of parse if any of your action handlers are async. Returns a Promise.

This way we wouldn't need to wrap each internal command with error handling logic, and just add a single try/catch around `cli.parseAsync` once.

I think with `cli.parse` it's not possible to catch the command execution error (it returns a promise instead of throwing), that's why `wrapCommand` was applied everywhere

Reply:
Yes, you're right, `cli.parse` actually executes command, however wrap around this call with try-catch handles error for external command only, but internal commands still needs its own wrapper :man_shrugging: 

Therefore, I decided to use `unhandledRejection` event, which fires for all commands.